src/main/java/uk/gov/hmcts/ccd/domain/service/search/elasticsearch/ElasticsearchQueryHelper.java

```diff
@@ -0,0 +1,159 @@
+
+    private final ObjectMapper objectMapper;
+    private final ApplicationParams applicationParams;
+    private final ObjectMapperService objectMapperService;
+    private final SearchQueryOperation searchQueryOperation;
+    private final GetCaseTypeOperation getCaseTypeOperation;
+    private final UserService userService;
+    private final ElasticsearchMappings elasticsearchMappings;
+
+    @Autowired
+    public ElasticsearchQueryHelper(@Qualifier("DefaultObjectMapper") ObjectMapper objectMapper,
+                                    ApplicationParams applicationParams,
+                                    ObjectMapperService objectMapperService,
+                                    SearchQueryOperation searchQueryOperation,
+                                    @Qualifier(AuthorisedGetCaseTypeOperation.QUALIFIER) GetCaseTypeOperation getCaseTypeOperation,
+                                    UserService userService,
+                                    ElasticsearchMappings elasticsearchMappings) {
+        this.objectMapper = objectMapper;
+        this.applicationParams = applicationParams;
+        this.objectMapperService = objectMapperService;
+        this.searchQueryOperation = searchQueryOperation;
+        this.getCaseTypeOperation = getCaseTypeOperation;
+        this.userService = userService;
+        this.elasticsearchMappings = elasticsearchMappings;
+    }
+
+    public CrossCaseTypeSearchRequest prepareRequest(List<String> caseTypeIds, String useCase, String jsonSearchRequest) {
+        rejectBlackListedQuery(jsonSearchRequest);
+
+        final List<String> updatedCaseTypeIds = buildCaseTypeIds(caseTypeIds);
+
+        JsonNode searchRequest = stringToJsonNode(jsonSearchRequest);
+        if (!Strings.isNullOrEmpty(useCase)) {
+            applyConfiguredSort(searchRequest, updatedCaseTypeIds, useCase);
+        }
+
+        return new CrossCaseTypeSearchRequest.Builder()
+            .withCaseTypes(updatedCaseTypeIds)
+            .withSearchRequest(searchRequest)
+            .build();
+    }
+
+    private void applyConfiguredSort(JsonNode searchRequest, List<String> caseTypeIds, String useCase) {
+        JsonNode sortNode = searchRequest.get(SORT);
+        if (sortNode == null) {
+            ArrayNode appliedSortsNode = buildSortNode(caseTypeIds, useCase);
+            if (appliedSortsNode.size() > 0) {
+                ((ObjectNode)searchRequest).set(SORT, appliedSortsNode);
+            }
+        }
+    }
+
+    private ArrayNode buildSortNode(List<String> caseTypeIds, String useCase) {
+        ArrayNode sortNode = objectMapper.createArrayNode();
+        caseTypeIds.forEach(caseTypeId -> addCaseTypeSorts(caseTypeId, useCase, sortNode));
+        return sortNode;
+    }
+
+    private void addCaseTypeSorts(String caseTypeId, String useCase, ArrayNode sortNode) {
+        Optional<CaseTypeDefinition> caseTypeOpt = getCaseTypeOperation.execute(caseTypeId, CAN_READ);
```

Comment:
temporary: wonder whether calling the getCaseTypeOperation is too "heavy". Perhaps would be ok to just call: 
    `CaseTypeDefinition getCaseType(String caseTypeId);` ?